docker/runner.go

```diff
@@ -22,52 +29,103 @@ func NewRunner() *Runner {
 }
 
 // Run the function at path
-func (n *Runner) Run(f bosonFunc.Function) error {
-	// Check for the docker binary explicitly so that we can return
-	// an extra-friendly error message.
-	_, err := exec.LookPath("docker")
+func (n *Runner) Run(ctx context.Context, f bosonFunc.Function) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 	if err != nil {
-		return errors.New("please install 'docker'")
+		return errors.Wrap(err, "failed to create docker api client")
 	}
 
-	if f.Image == "" {
-		return errors.New("Function has no associated Image. Has it been built?")
+	envs := make([]string, 0, len(f.EnvVars)+1)
+	for name, value := range f.EnvVars {
+		if !strings.HasSuffix(name, "-") {
+			envs = append(envs, fmt.Sprintf("%s=%s", name, value))
+		}
```

Comment:
I can't seem to remember why we need to explicitly exclude environment variables with a trailing dash.  Can you remind me (and perhaps add a comment and/or a test in case I forget again)?

Reply:
We need some way to tell an envvar was explicitly unset. Since envvar name must not contain dash, it can be (ab)used to indicate that envvar is unset (I took inspiration from `kn`).